tests/ui/unnecessary_result_map_or_else.fixed

```diff
@@ -0,0 +1,7 @@
+#![warn(clippy::unnecessary_result_map_or_else)]
+#![allow(clippy::unnecessary_literal_unwrap)]
+
+fn main() {
+    let x: Result<(), ()> = Ok(());
+    x.unwrap_or_else(|err| err); //~ ERROR: unused "map closure" when calling
```

Comment:
This looks like a good start. I'd like to see some tests where the closure

* ascribes a type
* auto-adjusts the type (e.g. auto-deref)
* uses a temporary variable (but still implements `id`)